### PR TITLE
Update project resolution and app settings writes in `dotnet user-jwts`

### PR DESCRIPTION
**Pull request by captainsafia**
- Updates `dotnet user-jwts` to use the same project finder logic as the `dotnet user-secrets` key
- Use `File.Create` to open write stream when writing file to open file with `FileMode.Create` and `FileShare.None` to support overwriting contents correctly

### CODE REVIEW

File: src/Tools/dotnet-user-jwts/src/Helpers/JwtAuthenticationSchemeSettings.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ public void Save(string filePath)
             };
         }
 
-        using var writer = new FileStream(filePath, FileMode.Open, FileAccess.Write);
+        using var writer = File.Create(filePath);
```
[the remark this answers is not in the document]
Comment (amcasey) [reply src/Tools/dotnet-user-jwts/src/Helpers/JwtAuthenticationSchemeSettings.cs:60]: What does it do by default?

[the remark this answers is not in the document]
Comment (BrennanConroy) [reply src/Tools/dotnet-user-jwts/src/Helpers/JwtAuthenticationSchemeSettings.cs:60]: https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs,2c80f62af94c907c,references

File: src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs [hunk 2]
```diff
@@ -80,6 +71,22 @@ public void Create_WritesGeneratedTokenToDisk()
...
+        var appsettings = Path.Combine(Path.GetDirectoryName(project), "appsettings.Development.json");
+        var app = new Program(_console);
+
+        app.Run(new[] { "create", "--project", project });
+        Assert.Contains("New JWT saved", _console.GetOutput());
+        var exception = Record.Exception(() => JsonSerializer.Deserialize<Dictionary<string, Jwt>>(File.ReadAllText(appsettings)));
+        Assert.Null(exception);
+        app.Run(new[] { "create", "--project", project, "--issuer", "new-issuer"  });
+        exception = Record.Exception(() => JsonSerializer.Deserialize<Dictionary<string, Jwt>>(File.ReadAllText(appsettings)));
+        Assert.Null(exception);
```
[the remark this answers is not in the document]
Comment (amcasey) [reply src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs:87]: It does seem odd to deserialize a dictionary and then not look at its contents.

File: src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs [hunk 3]
```diff
@@ -80,6 +71,22 @@ public void Create_WritesGeneratedTokenToDisk()
...
+    [Fact]
+    public void Create_CanModifyExistingScheme()
+    {
+        var project = Path.Combine(_fixture.CreateProject(), "TestProject.csproj");
+        var appsettings = Path.Combine(Path.GetDirectoryName(project), "appsettings.Development.json");
+        var app = new Program(_console);
+
+        app.Run(new[] { "create", "--project", project });
+        Assert.Contains("New JWT saved", _console.GetOutput());
+        var exception = Record.Exception(() => JsonSerializer.Deserialize<Dictionary<string, Jwt>>(File.ReadAllText(appsettings)));
```
Comment (amcasey) [inline src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs:83]: I'm sure this follows an existing pattern, but why not just let it throw?

File: src/Tools/dotnet-user-jwts/src/Helpers/JwtAuthenticationSchemeSettings.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (amcasey) [reply src/Tools/dotnet-user-jwts/src/Helpers/JwtAuthenticationSchemeSettings.cs:60]: Oh, okay, you're suggesting we restrict permissions. Seems reasonable.

[the remark this answers is not in the document]
Comment (amcasey) [reply src/Tools/dotnet-user-jwts/src/Helpers/JwtAuthenticationSchemeSettings.cs:60]: While we're at it, `FileMode.Open` and `FileMode.Create` are pretty different - was that change intentional?

File: src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs [hunk 4]
```diff
@@ -80,6 +71,22 @@ public void Create_WritesGeneratedTokenToDisk()
         Assert.Contains("dotnet-user-jwts", File.ReadAllText(appsettings));
     }
 
+    [Fact]
+    public void Create_CanModifyExistingScheme()
```
Comment (amcasey) [inline src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs:75]: For my own edification, this has nothing to do with the product change, right?

File: src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs [hunk 5]
```diff
@@ -631,7 +638,19 @@ public void List_CanHandleNoProjectOptionProvided_WithNoProjects()
         var app = new Program(_console);
         app.Run(new[] { "list" });
 
-        Assert.Contains("No project found at `-p|--project` path or current directory.", _console.GetOutput());
+        Assert.Contains($"Could not find a MSBuild project file in '{Directory.GetCurrentDirectory()}'. Specify which project to use with the --project option.", _console.GetOutput());
+    }
+
+    [Fact]
+    public void List_CanHandleProjectOptionAsPath()
+    {
```
Comment (amcasey) [inline src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs:646]: I think this is also unrelated to the project change?

File: src/Tools/dotnet-user-jwts/src/Helpers/JwtAuthenticationSchemeSettings.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (captainsafia) [reply src/Tools/dotnet-user-jwts/src/Helpers/JwtAuthenticationSchemeSettings.cs:60]: Yep, the `File.Create` sets the correct options for being able to overwrite the file.



> We should probably be a good citizen and use `FileStream` with `FileStreamOptions` and set the [UnixCreateMode](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/IO/FileStreamOptions.cs,128) to `UserRead | UserWrite`

I had considered initializing a FileStream with the more detailed permissions but it felt bloated compared to the brevity of `File.Create`. I'm fine with modifying this though.

File: src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs (hunk 3 above)
Comment (captainsafia) [reply src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs:83]: We are trying to verify that it does not throw. This test validates that the previous behavior, where in calling the CLI command twice would write invalid JSON to the file, now longer occurs. Similarly, this is why this test case doesn't validate the actual contents of the file since there are other tests that cover that.

File: src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs (hunk 5 above)
Comment (captainsafia) [reply src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs:646]: It is. Previously, we did not support searching for projects given a directory, we expected a complete filepath to a csproj. This allows querying with a filepath as the project options. The test name could probably be a little clearer.

File: src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs (hunk 3 above)
Comment (amcasey) [reply src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs:83]: Wouldn't the test fail if that call threw?  Is that different from it failing on the next line when we assert that the exception is null?

> Similarly, this is why this test case doesn't validate the actual contents of the file since there are other tests that cover that.

I'm not sure I follow.  There's another test that updates the scheme and validates that the new one is in place?  If so, why have this test at all?  If not, isn't it worthwhile to confirm that the update actually took effect?

**Review by BrennanConroy — APPROVED** [review]

File: src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs (hunk 3 above)
Comment (captainsafia) [reply src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs]: > Wouldn't the test fail if that call threw? Is that different from it failing on the next line when we assert that the exception is null?

Yes, the test would fail if the call threw but this is a more explicit way of clarifying aht we don't expect it to through.

> There's another test that updates the scheme and validates that the new one is in place? If so, why have this test at all? If not, isn't it worthwhile to confirm that the update actually took effect?

I double checked and it looks like we don't have a test that confirms these updates yet so I've updated the code change to match it appropriately.
